rule/context_keys_type.go

```diff
@@ -24,7 +27,9 @@ func (*ContextKeysType) Apply(file *lint.File, _ lint.Arguments) []lint.Failure
 		},
 	}
 
-	file.Pkg.TypeCheck()
+	if err := file.Pkg.TypeCheck(); err != nil {
+		r.logger.Printf("Rule=%q TypeCheck() error=%v\n", r.Name(), err)
+	}
```

Comment:
Something else I don't understand.

Maybe my feedback is naive. So please bare with me.

file.Pkg.TypeChecked is called here, but apparently it is called in all rules (at least the one you modify)

We were ignoring the error, and now we log it.

But Apply receive `file` as argument, so what about calling the file.Pkg.TypeCheck in the code that calls Apply.

So we wouldn't need adding a logger everywhere.

Maybe, calling file.Pkg.TypeCheck was not needed in all rules, but is it a real problem to call it?

Reply:
We are logging an error to let the user know that something is wrong without changing previous behaviour.

Reply:
Let's consider this thread as resolved

More information can be found in another thread

https://github.com/mgechev/revive/pull/1278#discussion_r2019778786